src/js/components/NetworkErrorNotice.tsx

```diff
@@ -15,21 +16,39 @@ const MAX_BACKOFF = 128
 
 export default function NetworkErrorNotice({error}: Props) {
   const dispatch = useDispatch()
-  const spaceId = useSelector(Current.getSpaceId)
+  const connId = useSelector(Current.getConnectionId)
+  const status = useSelector(ConnectionStatuses.get(connId))
   const [count, setCount] = useState(0)
+  const statusRef = useRef(status)
+  const connRef = useRef(connId)
 
-  const retry = () => dispatch(initSpace(spaceId))
-  const dismiss = () => dispatch(Notice.dismiss())
+  const retry = () => {
+    setCount(0)
+    dispatch(checkStatus())
+  }
+  const dismiss = () => {
+    dispatch(Notice.dismiss())
+  }
 
   useEffect(() => {
+    if (connId === connRef.current) {
+      statusRef.current = status
+      if (status === "disconnected") dismiss()
+    }
+  }, [status, connId])
+
+  useEffect(() => {
+    connRef.current = connId
+    dispatch(ConnectionStatuses.set(connId, "retrying"))
```

Comment:
on first render, the component sets the connId, which we will continue to refer to in the above effect so we are sure that we are only managing the connection in question. we then set it's status to retrying.